### ISSUE TITLE
Feature request: Selection match and replace, a.k.a. passive mode

### ISSUE THREAD

**Issue by telmotrooper** [federico-terzi/espanso#62]
Hello, thank you for your work on espanso (the site looks quite professional too).

While I do like the application as is, I don't quite like having to wait for the application to replace my match as soon as I type it.

Would you consider adding a "passive" mode, where I run espanso over a text selection and only then it replaces the matches it finds? It could be triggered with a hotkey for example.

I also prefer that approach because the application won't keep listening to all my keyboard input (and I am aware I can disable it anytime with alt + alt).

Thank you very much and a have a good day! 😊️

**Comment by federico-terzi** [comment]
Hi, thanks a lot for the feedback!
Ok so let's formalize this idea. Please tell me if this is what you mean:

### Passive Mode Proposal
Changes: Avoid replacing the match as soon as the user finishes typing it.
Actions required:
* The user selects some text containing a keyword
* The user press a keyboard shortcut ( for example CTRL + CTRL )
* If the selected text matches a keyword, espanso replaces it.

Is this what you mean? 

The user would have to select the keyword every time though, and this could be an inconvenience.

An alternative would be to avoid the text selection but wait until the keyboard shortcut to make the replacement. Let me know :)

**Comment by telmotrooper** [comment]
Mostly yes, but I actually mean for the user to select the whole text block that contains the matches.

**Example:** 

The user is writing a message, he then uses Ctrl + A to select the whole text block:
```
Today is :date
:br
```
When he presses Ctrl + Ctrl (as you suggested) the text becomes:
```
Today is 10/03/2019
Best Regards,
Telmo "Trooper"
```

Of course if the code block is too big it will probably take longer for the application to run, but in such cases the user could select partial chunks of text.

**Comment by federico-terzi** [comment]
Ok, I finally understood your idea. It's pretty good, thanks for the suggestion.
I will definitely add it to the Roadmap!

**Comment by beefeater7** [comment]
I support the suggestion. I like to work with the raw input, then feed it to the parser. Like coding.

**Comment by federico-terzi** [comment]
Thanks for the feedback @beefeater7. I'll try to add it in the next major version of espanso :)
I can't promise when it will come though, as I'm pretty busy with university at the moment

**Comment by federico-terzi** [comment]
Hey,
I'm closing this issue as this feature has been shipped in version 0.5.0.
Cheers :)
